lib/turkish_cities/district.rb

```diff
@@ -8,62 +8,66 @@
 class District
   include DecomposerHelper
 
-  def list_subdistricts(city_name, district_name)
-    district_list = create_district_list(city_name)
+  def initialize(city_name, district_name)
+    @city_name = city_name
+    @district_name = district_name
+    @district_list = create_district_list(city_name)
+  end
 
-    if !district_list[district_name].nil?
+  def list_subdistricts
```

Comment:
burayi belki soyle yapabiliriz.

```ruby
def list_subdistricts
  district_not_found_error(@district_name, @city_name) unless district_item.nil?
  
  subdistricts = []
  district_item.each do |subdistrict|
    subdistricts << subdistrict[0]
  end
  
  sort_alphabetically(subdistricts)
end
```

ama daha da guzel olabilir. `each` block'u daha iyi yazilabilir geldi ancak `district_item` nasil birsey donuyor pek anlamadim.

Reply:
`district_item` string'lerden olusan bir array donebiliyor. 
each block'u nasil daha iyi yazilabilir aklima bir sey gelmedi acikcasi. Senin yazdigin sekilde iyi gorunuyor bence ama ilk satirda unless yerine if kullanmak gerekiyor: 
`district_not_found_error(@district_name, @city_name) if district_item.nil?`

Reply:
Baştan error kontrolü yapabildiğimiz yerlerde dediğiniz gibi yapabiliriz. Serpil'in dediği gibi unless yerine if olduğunda tam oluyor. Bir tek 'create_neighborhoods' da kullanamayız bu bloku. Orada çok değişik if ler var.